cluster-autoscaler/simulator/cluster_snapshot_test.go

```diff
@@ -464,3 +463,126 @@ func TestNodeAlreadyExists(t *testing.T) {
 		}
 	}
 }
+
+func TestPVCUsedByPods(t *testing.T) {
+	node := BuildTestNode("node", 10, 10)
+	pod1 := BuildTestPod("pod1", 10, 10)
+	pod1.Spec.NodeName = node.Name
+	pod1.Spec.Volumes = []apiv1.Volume{
+		{
+			Name: "v1",
+			VolumeSource: apiv1.VolumeSource{
+				PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "claim1",
+				},
+			},
+		},
+	}
+	pod2 := BuildTestPod("pod2", 10, 10)
+	pod2.Spec.NodeName = node.Name
+	pod2.Spec.Volumes = []apiv1.Volume{
+		{
+			Name: "v1",
+			VolumeSource: apiv1.VolumeSource{
+				PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "claim1",
+				},
+			},
+		},
+		{
+			Name: "v1",
+			VolumeSource: apiv1.VolumeSource{
+				PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "claim2",
+				},
+			},
+		},
+	}
+	nonPvcPod := BuildTestPod("pod1", 10, 10)
+	nonPvcPod.Spec.NodeName = node.Name
+	nonPvcPod.Spec.Volumes = []apiv1.Volume{
+		{
+			Name: "v1",
+			VolumeSource: apiv1.VolumeSource{
+				NFS: &apiv1.NFSVolumeSource{
+					Server:   "",
+					Path:     "",
+					ReadOnly: false,
+				},
+			},
+		},
+	}
+	testcase := []struct {
+		desc              string
+		node              *apiv1.Node
+		pods              []*apiv1.Pod
+		claimName         string
+		exists            bool
+		removePod         string
+		existsAfterRemove bool
+	}{
+		{
+			desc:      "pvc new pod with volume fetch",
+			node:      node,
+			pods:      []*apiv1.Pod{pod1},
+			claimName: "claim1",
+			exists:    true,
+			removePod: "",
+		},
+		{
+			desc:      "pvc new pod with incorrect volume fetch",
+			node:      node,
+			pods:      []*apiv1.Pod{pod1},
+			claimName: "incorrect-claim",
+			exists:    false,
+			removePod: "",
+		},
+		{
+			desc:      "new pod with non-pvc volume fetch",
+			node:      node,
+			pods:      []*apiv1.Pod{nonPvcPod},
+			claimName: "incorrect-claim",
+			exists:    false,
+			removePod: "",
+		},
+		{
+			desc:              "pvc new pod with delete volume fetch",
+			node:              node,
+			pods:              []*apiv1.Pod{pod1},
+			claimName:         "claim1",
+			exists:            true,
+			removePod:         "pod1",
+			existsAfterRemove: false,
+		},
+		{
+			desc:              "pvc two pods with duplicated volume, delete one pod, fetch",
```

Comment:
I feel like some test cases with multiple insertions and deletions, including both pods using and not using a pvc would also be good.

Reply:
Done.